### **Limpieza de datos**

In [5]:
# Cargar las librerias necesarias
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [6]:
# Cargar el dataset 
df_tortilla_prices = pd.read_csv('../data/raw/tortilla_prices.csv')
df_tortilla_prices.head(15)

State              City  Year  Month  Day  \
0        Aguascalientes    Aguascalientes  2007      1   10   
1       Baja California          Mexicali  2007      1   10   
2       Baja California           Tijuana  2007      1   10   
3   Baja California Sur            La Paz  2007      1   10   
4              Campeche          Campeche  2007      1   10   
5              Coahuila    Piedras Negras  2007      1   10   
6              Coahuila          Saltillo  2007      1   10   
7                Colima            Colima  2007      1   10   
8               Chiapas         Tapachula  2007      1   10   
9               Chiapas  Tuxtla Gutiérrez  2007      1   10   
10            Chihuahua         Chihuahua  2007      1   10   
11            Chihuahua        Cd. Juárez  2007      1   10   
12                 D.F.              D.F.  2007      1   10   
13                 D.F.           ZM D.F.  2007      1   10   
14              Durango           Durango  2007      1   10   

           Store type  Price per kilogram  
0   Mom and Pop Store                9.90  
1   Mom and Pop Store                 NaN  
2   Mom and Pop Store               10.00  
3   Mom and Pop Store               10.00  
4   Mom and Pop Store               10.00  
5   Mom and Pop Store               10.00  
6   Mom and Pop Store                 NaN  
7   Mom and Pop Store               10.00  
8   Mom and Pop Store                 NaN  
9   Mom and Pop Store                9.20  
10  Mom and Pop Store               10.00  
11  Mom and Pop Store                 NaN  
12  Mom and Pop Store                8.33  
13  Mom and Pop Store                7.67  
14  Mom and Pop Store                7.10

##### **Eliminar valores nulos**

In [ ]:
# Verificar la cantidad de fidas y columnas 
num_fias, num_columnas = df_tortilla_prices.shape
print(f'Número de filas: {num_fias}\nNúmero de columnas: {num_columnas}')

Número de filas: 289146
Número de columnas: 7


In [8]:
# Mostrar las filas con los valores nulos
filas_con_nulos = df_tortilla_prices[df_tortilla_prices.isnull().any(axis=1)]
filas_con_nulos

State                   City  Year  Month  Day  \
1       Baja California               Mexicali  2007      1   10   
6              Coahuila               Saltillo  2007      1   10   
8               Chiapas              Tapachula  2007      1   10   
11            Chihuahua             Cd. Juárez  2007      1   10   
15              Durango          Gómez Palacio  2007      1   10   
...                 ...                    ...   ...    ...  ...   
271853           Sonora  San Luis Río Colorado  2023      6   21   
271907           Sonora  San Luis Río Colorado  2023      6   23   
271961           Sonora  San Luis Río Colorado  2023      6   26   
276528       Guanajuato                   León  2024      1   17   
283008       Guanajuato                   León  2024      1   17   

               Store type  Price per kilogram  
1       Mom and Pop Store                 NaN  
6       Mom and Pop Store                 NaN  
8       Mom and Pop Store                 NaN  
11      Mom and Pop Store                 NaN  
15      Mom and Pop Store                 NaN  
...                   ...                 ...  
271853   Big Retail Store                 NaN  
271907   Big Retail Store                 NaN  
271961   Big Retail Store                 NaN  
276528  Mom and Pop Store                 NaN  
283008   Big Retail Store                 NaN  

[6390 rows x 7 columns]

In [14]:
# Eliminar las filas con los valores nulos
df_tortilla_prices_sin_nulos = df_tortilla_prices.dropna()

# Verificar la cantidad de filas y columnas
num_filas_sin_nulos, num_columnas_sin_nulos = df_tortilla_prices_sin_nulos.shape
print(f'Número de filas: {num_filas_sin_nulos}\nNúmero de columnas: {num_columnas_sin_nulos}')

Número de filas: 282756
Número de columnas: 7


In [18]:
# Verificar si quedan valores nulos
num_nulos = df_tortilla_prices_sin_nulos.isnull().sum()
print(f'Números de valores nulos:\n{num_nulos}')

Números de valores nulos:
State                 0
City                  0
Year                  0
Month                 0
Day                   0
Store type            0
Price per kilogram    0
dtype: int64


Antes de eliminar los valores nulos el DataFrame contenia 289146 orginalmente.

Despues de eliminar los valores nulos el DataFrame contiene 282756.

Se eliminaron 6390 filas con valores nulos.